### ISSUE TITLE
WasmStripILAfterAOT causes broken runtime/compiler - System.InvalidProgramException: Common Language Runtime detected an invalid program

### ISSUE THREAD

**Issue by Matheos96** [dotnet/runtime#98041]
### Description

Sorry if this is in the wrong project, please move it if it is.

.NET 8 
SDK: 8.0.101
Runtimes: 8.0.1

Today I played around with Blazor WASM (Standalone) AOT compilation. It looked promising and I finally enabled the `<WasmStripILAfterAOT>true</WasmStripILAfterAOT>`  to see if I can gain some size reduction. I had trouble compiling after enablng this flag.. I can't remember the exact error but I believe it came from the Program Files\dotnet folder and it was shouting to me about Permission issues regarding MOVE of some files..?! I thought that it must know what it is doing so I re-launched my IDE (Jetbrains Rider) and re-initiateted the publish process. It succeeded but the wasm app did not work. I faced this issue: https://github.com/dotnet/runtime/issues/92238. Anyway, that is not my problem exactly. I just thought that I will simply not use this flag until it is fixed, no big deal. WRONG...

After that compilation I can no longer launch our ASP.NET Web project (API) at all anymore. It is completely separate from the wasm project and should not be affected by it at all. Nor can I of course launch the Blazor project anymore... 

I keep getting ` System.InvalidProgramException: Common Language Runtime detected an invalid program` from a seemingly unrelated random line in the ASP.NET project:
![image](https://github.com/dotnet/runtime/assets/32064310/067e9041-22a8-4675-a33a-be8bf8cd5da6)

The blazor wasm app gives the following errors:
![image](https://github.com/dotnet/runtime/assets/32064310/6e514f32-12fb-45c3-8cbb-ed6a3911f004)

```
Error: [MONO] /__w/1/s/src/mono/mono/mini/interp/interp.c:3629 <disabled> at ht (https://localhost:5011/_framework/dotnet.runtime.8.0.1.uyhyezbwwb.js:3:12765) at Ul (https://localhost:5011/_framework/dotnet.runtime.8.0.1.uyhyezbwwb.js:3:175673) at https://localhost:5011/_framework/dotnet.native.wasm:wasm-function[120]:0xa330 at https://localhost:5011/_framework/dotnet.native.wasm:wasm-function[634]:0x3f331 at https://localhost:5011/_framework/dotnet.native.wasm:wasm-function[551]:0x3bde5 at https://localhost:5011/_framework/dotnet.native.wasm:wasm-function[553]:0x3bea8 at https://localhost:5011/_framework/dotnet.native.wasm:wasm-function[554]:0x3bedb at https://localhost:5011/_framework/dotnet.native.wasm:wasm-function[239]:0x1b3d0 at https://localhost:5011/_framework/dotnet.native.wasm:wasm-function[273]:0x1d1b1 at https://localhost:5011/_framework/dotnet.native.wasm:wasm-function[3185]:0xe81aa

Uncaught (in promise) Error: Failed to start platform. Reason: [object Object]

```

I am able to BUILD projects without errors but I cannot run the projects produced by my builds. I strongly suspect that there is something during the compilation process to IL code that messes up my output. 

I tried building + running the ASP.NET app in WSL and that worked fine! No issues at all. So basically something on my Windows install is really messed up and I have no clue even where to look next...

### Reproduction Steps

Not sure if easily reproducable. I just enabled the `<WasmStripILAfterAOT>true</WasmStripILAfterAOT>` flag in our Blazor wasm .net 8 standalone app .csproj (+ AOT enabled) and published as self contained portable using Jetbrains Rider on Windows 11 x64 while running the IDE as Administrator.

### Expected behavior

`<WasmStripILAfterAOT>true</WasmStripILAfterAOT>` does not make my work PC unusable for .NET development....

### Actual behavior

My work computers .NET environment is completely broken and I cannot figure out how to fix it. I have tried everything I can think of:

- Reinstalled .NET core versions
- cleaned obj and bin folders
- clean check out of our gitlab repository
- Uninstalled EVERYTHING .NET related (even windows enable/disable features). All .NET SDKs, all .NET runtimes, ALL .NET Framework stuff EVERYTHING from Visual Studio (all components) all Visual Studio versions, manualy dotnet folder deletion and Microsoft Visual Studio folder deletions. Running Visual Studio Installer cleanup exe
- Rebooted after every tried step
- uninstalled/reinstalled dotnet workloads (if not a given)
- Basically I have cleaned my PC of everything .NET Core/Framework related - even C++ redists... (desperate) 

### Known Workarounds

Builds and works on other machines. Even in WSL on the same machine with the exact same code base

### Configuration

- NET 8 (8.0.1 and 8.0.101)
- Windows 11
- x64
- Only occurs on my machine
- Blazor yes - but browser is surely not relevant as the issue is much deeper. I use chrome

**Comment by Matheos96** [comment]
After 5 hours of pulling my hair I figured it out (not five hours since my post but I was well into it when I created the post).

The compilation process had seemingly messed something up in my .nuget/packages folder. Probably nowhere else... All I had to do was delete the folder and re-run nuget package restores and now everything works again...
I did also have to install wasm-tools workload obviously to get the blazor app working again but now everything is running smoothly...

5 hours of my working day (or working day + 4 hours) wasted... I still feel this is a major issue that should never occur. How such a small thing as a .csproj flag can mess up Nuget package cache dll:s is beyond me. I am not touching that flag again until it is fixed.... 

Leaving this open for now so that it surely get's some attention. Feel free to close it once reviewed

**Comment by ennerperez** [comment]
In general the AOT causes broken runtime/compiler - System.InvalidProgramException: Common Language Runtime detected an invalid program. Not only Wasm, NativeAOT in MAUI + Trim

**Comment by fanyang-mono** [comment]
I apologize for all you've gone through. This issue should have been resolved in .NET9 since preview 1. Please download the latest .NET9 from [here](https://dotnet.microsoft.com/en-us/download/dotnet/9.0) and try it out.

**Comment by fanyang-mono** [comment]
That's good to hear! Hope that the size reduction is helpful for your project. Closing this issue now.
